### ISSUE TITLE
Indent works incorrectly with multiline arguments

### ISSUE THREAD

**Issue by polomsky** [eslint/eslint#9706]
**Tell us about your environment**

* **ESLint Version:** 4.13.0
* **Node Version:** 8.9.3
* **npm Version:** 5.5.1

**What parser (default, Babel-ESLint, etc.) are you using?**
default
**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
    "rules": {
        "indent": [
            "error",
            4,
            {
                "SwitchCase": 1
            }
        ],
    }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
'use strict';

function foo() {

}

foo("A" +
    "B",
    "C"); // problematic indents
```

```bash
eslint test.js
```

**What did you expect to happen?**
File will be valid
**What actually happened? Please include the actual, raw output from ESLint.**
```
  9:1  error  Expected indentation of 0 spaces but found 4  indent

✖ 1 problem (1 error, 0 warnings)
  1 error, 0 warnings potentially fixable with the `--fix` option.
```

**Comment by platinumazure** [comment]
@not-an-aardvark Any thoughts?

**Comment by not-an-aardvark** [comment]
Looks like a bug to me.

**Comment by not-an-aardvark** [comment]
After looking into it more, this is happening for the same reason as the following case, which we currently consider to be working as intended because it seems to be common.

```js
foo({
  a: 1,
},
{
  b: 2
});
```

In both cases, the first argument is multiline and starts on the same line as the function call. (The same logic is used for arrays, objects, blocks, parameters, imports, etc.) I'm not sure if there's a good place to draw the line here, so this might be considered "works-as-intended" unless we can determine a better way to enforce the desired behavior in both cases.

**Comment by nzakas** [comment]
Unfortunately, it looks like there wasn't enough interest from the team or community to implement this change. While we wish we'd be able to accommodate everyone's requests, we do need to prioritize. We've found that issues failing to be implemented after 90 days tend to never be implemented, and as such, we [close those issues](https://eslint.org/docs/maintainer-guide/issues#when-to-close-an-issue). This doesn't mean the idea isn't interesting or useful, just that it's not something the team can commit to.
